subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/DefaultXcodeProject.java

```diff
@@ -16,30 +16,53 @@
 
 package org.gradle.ide.xcode.internal;
 
+import org.gradle.api.file.FileCollection;
 import org.gradle.ide.xcode.XcodeProject;
+import org.gradle.ide.xcode.internal.xcodeproj.GidGenerator;
 
 import java.io.File;
 import java.util.ArrayList;
-import java.util.HashSet;
+import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
 public class DefaultXcodeProject implements XcodeProject {
-    private final List<XcodeTarget> targets = new ArrayList<XcodeTarget>();
-    private final List<XcodeScheme> schemes = new ArrayList<XcodeScheme>();
-    private final Set<File> sources = new HashSet<File>();
+    private final GidGenerator gidGenerator = new GidGenerator(Collections.<String>emptySet());
+    private final List<File> additionalFiles = new ArrayList<File>();
+    private XcodeTarget target;
+    private FileCollection sources;
```

Comment:
I would make `sources` a `ConfigurableFileCollection` instance and add the additional files to this instance rather than maintaining a separate collection.